### PR TITLE
Add a KVO to-many notification test for aggregate functions.

### PR DESCRIPTION
**Pull request by DHowett-MSFT**
This was written with/for the KVO to-many collection feature, but it didn't make it in because we did not, at the time, support aggregate functions.

### CODE REVIEW

**Comment by DHowett-MSFT** [comment]
@msft-Jeyaram is added to the review. #Closed

**Comment by DHowett-MSFT** [comment]
@MSFTFox is added to the review. #Closed

**Comment by rajsesh-msft** [comment]
:shipit:

File: tests/unittests/Foundation/NSObject_KeyValueObservationTests.mm [hunk 1]
```diff
@@ -362,4 +362,40 @@ - (NSUInteger)hits {
...
+                     // This array is assisted by setter functions, and should also dispatch one notification per change.
+                     NSMutableArray* mediatedVersionOfArray = [observee mutableArrayValueForKey:@"arrayWithHelpers"];
+                     [mediatedVersionOfArray addObject:@"object1"];
+                 }
+        andExpectChangeCallbacks:@[ insertCallbackPost, illegalChangeNotification ]];
+    EXPECT_EQ(1, facade.hits);
+
+    facade = [[TEST_IDENT(Facade) new] autorelease];
+    // In this test, we use the same arrayWithHelpers as above, but interact with it manually.
+    [facade observeKeyPath:@"arrayWithHelpers.@count"
```
Comment (msft-Jeyaram) [inline tests/unittests/Foundation/NSObject_KeyValueObservationTests.mm]: should we add a test for an aggregate  function with property? e.g max,avg.

**Comment by msft-Jeyaram** [comment]
:shipit:

File: tests/unittests/Foundation/NSObject_KeyValueObservationTests.mm (hunk 1 above)
Comment (DHowett-MSFT) [inline tests/unittests/Foundation/NSObject_KeyValueObservationTests.mm]: It's not possible to notify on an aggregate function with a sub-property on the reference platform :)

**Comment by mnithish** [comment]
:shipit:

**Comment by DHowett-MSFT** [comment]
Thanks everyone; in CI build now.

**Pull request merged**